lib/ret_web/views/api/v1/avatar_view.ex

```diff
@@ -0,0 +1,43 @@
+defmodule RetWeb.Api.V1.AvatarView do
+  use RetWeb, :view
+  alias Ret.OwnedFile
+
+  defp url_for_avatar(avatar) do
+    "#{RetWeb.Endpoint.url()}/avatars/#{avatar.avatar_sid}/#{avatar.slug}"
+  end
+
+  def render("create.json", %{avatar: avatar}) do
+    %{avatars: [render_avatar(avatar)]}
+  end
+
+  def render("show.json", %{avatar: avatar}) do
+    %{avatars: [render_avatar(avatar)]}
+  end
+
+  defp file_url_or_nil(avatar, column) do
+    case avatar |> Map.get(column) do
+      nil -> nil
+      owned_file -> owned_file |> OwnedFile.uri_for() |> URI.to_string()
+    end
+  end
+
+  def render_avatar(avatar) do
+    %{
+      avatar_id: avatar.avatar_sid,
+      parent_avatar_id: unless(is_nil(avatar.parent_avatar), do: avatar.parent_avatar.avatar_sid),
+      name: avatar.name,
+      description: avatar.description,
+      attributions: if(is_nil(avatar.attributions), do: [], else: avatar.attributions),
+      allow_remixing: avatar.allow_remixing,
+      allow_promotion: avatar.allow_promotion,
+      files: %{
```

Comment:
you could refactor this into a list comprehension but it might turn out to be uglier